### PR TITLE
Add Third Party CSI Driver Chapter

### PR DESCRIPTION
**Pull request by futuretea**
**Related issues**
https://github.com/harvester/harvester/issues/4420

- Add Third Party CSI Driver Chapter
- Update `os.after_install_chroot_commands`
- Change onBrokenLinks from throw to warn to skip the missing of zh docs

### CODE REVIEW

File: docs/advanced/csidriver.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/dev/advanced/csidriver"/>
+</head>
+
+
+Harvester now offers the capability to install a [Container Storage Interface (CSI)](https://kubernetes-csi.github.io/docs/introduction.html) in your Harvester cluster. This allows you to leverage external storage for the Virtual Machine's non-system data disk, giving you the flexibility to use different drivers tailored for specific needs, whether it's for performance optimization or seamless integration with your existing in-house storage solutions.
+
+:::note
+
+The provisioner for the Virtual Machine (VM) image in Harvester still relies on Longhorn. Prior to version 1.2.0, Harvester exclusively supported Longhorn for storing VM data and did not offer support for external storage as a destination for VM data.
```
Comment (LucasSaintarbor) [inline docs/advanced/csidriver.md:16]:
```suggestion
The Virtual Machine (VM) image provisioner in Harvester still relies on Longhorn. Before version 1.2.0, Harvester exclusively supported Longhorn for storing VM data and did not offer support for external storage as a destination for VM data.
```

File: docs/advanced/csidriver.md [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+sidebar_label: Third-Party CSI Driver
+title: "Third-Party CSI Driver"
+---
+
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/dev/advanced/csidriver"/>
+</head>
+
+
+Harvester now offers the capability to install a [Container Storage Interface (CSI)](https://kubernetes-csi.github.io/docs/introduction.html) in your Harvester cluster. This allows you to leverage external storage for the Virtual Machine's non-system data disk, giving you the flexibility to use different drivers tailored for specific needs, whether it's for performance optimization or seamless integration with your existing in-house storage solutions.
```
Comment (LucasSaintarbor) [inline docs/advanced/csidriver.md:12]:
```suggestion
Harvester now offers the capability to install a [Container Storage Interface (CSI)](https://kubernetes-csi.github.io/docs/introduction.html) in your Harvester cluster. This allows you to leverage external storage for the Virtual Machine's non-system data disk, allowing you to use different drivers tailored for specific needs, whether for performance optimization or seamless integration with your existing in-house storage solutions.
```

File: docs/advanced/csidriver.md [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+:::
+
+## Prerequisites
+
+For the Harvester functions to work well, the third-party CSI driver needs to have the following capabilities:
+- Support expansion
+- Support snapshot
+- Support clone
+- Support block device
+- Support Read-Write-Many(RWX), for [Live Migration](../vm/live-migration.md)
```
Comment (LucasSaintarbor) [inline docs/advanced/csidriver.md:27]:
```suggestion
- Support Read-Write-Many (RWX), for [Live Migration](../ vm/live-migration.md)
```

File: docs/advanced/csidriver.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+
+## Create Harvester cluster
+
+Harvester's operating system follows an immutable design, which means that most files in the OS revert to their pre-configured state after a reboot.
+Therefore, you might need to perform additional configurations before installing the Harvester cluster for third-party CSI drivers.
```
Comment (LucasSaintarbor) [inline docs/advanced/csidriver.md:32]:
```suggestion
Harvester's operating system follows an immutable design, meaning that most OS files revert to their pre-configured state after a reboot. Therefore, you might need to perform additional configurations before installing the Harvester cluster for third-party CSI drivers.
```

File: docs/advanced/csidriver.md [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+
+:::note
+
+Upgrading Harvester causes the changes to the OS in the `after-install-chroot` stage to be lost. You must also configure the `after-upgrade-chroot` to make your changes persistent across an upgrade. Refer to [Runtime persistent changes](https://rancher.github.io/elemental-toolkit/docs/customizing/runtime_persistent_changes/) before upgrading Harvester.
+
+:::
+
+## Install the CSI driver
+
+After the cluster installed, refer to [How can I access the kubeconfig file of the Harvester cluster?](../faq.md#how-can-i-access-the-kubeconfig-file-of-the-harvester-cluster) to get the kubeconfig of the Harvester cluster.
```
Comment (LucasSaintarbor) [inline docs/advanced/csidriver.md:46]:
```suggestion
After installing the Harvester cluster is complete, refer to [How can I access the kubeconfig file of the Harvester cluster?](../faq.md#how-can-i-access-the-kubeconfig-file-of-the-harvester-cluster) to get the kubeconfig of the cluster.
```

File: docs/advanced/csidriver.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+With the kubeconfig of the Harvester cluster, you can install the third-party CSI drivers into the Harvester cluster by following the installation instructions for each CSI driver.
+
+And you also need to refer to the CSI driver documentation to create the StorageClass and VolumeSnapshotClass in the Harvester cluster.
```
Comment (LucasSaintarbor) [inline docs/advanced/csidriver.md:50]:
```suggestion
With the kubeconfig of the Harvester cluster, you can install the third-party CSI drivers into the cluster by following the installation instructions for each CSI driver. You must also refer to the CSI driver documentation to create the StorageClass and VolumeSnapshotClass in the Harvester cluster.
```

File: docs/advanced/csidriver.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+## Configure Harvester Cluster
+
+Before you can make use of Harvester's **Backup & Snapshot** features, some essential configurations need to be set up through the Harvester [csi-driver-config](../advanced/settings.md#csi-driver-config) setting. Here's how to do it:
```
Comment (LucasSaintarbor) [inline docs/advanced/csidriver.md:54]:
```suggestion
Before you can make use of Harvester's **Backup & Snapshot** features, you need to set up some essential configurations through the Harvester [csi-driver-config](../advanced/settings.md#csi-driver-config) setting. Follow these steps to make these configurations: 
```

File: docs/advanced/csidriver.md [hunk 8]
```diff
@@ -0,0 +1,66 @@
...
+
+1. Login to the Harvester UI, then navigate to **Advanced** > **Settings**.
+2. Find and select **csi-driver-config**, and then click on the **⋮** > **Edit Setting** to access the configuration options.
+3. In the settings, set the **Provisioner** to the third-party CSI driver.
+4. Next, Configure the **Volume Snapshot Class Name**. This setting points to the name of the `VolumeSnapshotClass` used for creating volume snapshots or VM snapshots.
+5. Similarly, Configure the **Backup Volume Snapshot Class Name**. This corresponds to the name of the `VolumeSnapshotClass` responsible for creating VM backups.
```
Comment (LucasSaintarbor) [inline docs/advanced/csidriver.md:60]:
```suggestion
1. Login to the Harvester UI, then navigate to **Advanced** > **Settings**.
1. Find and select **csi-driver-config**, and then select **⋮** > **Edit Setting** to access the configuration options.
1. Set the **Provisioner** to the third-party CSI driver in the settings.
1. Next, Configure the **Volume Snapshot Class Name**. This setting points to the name of the `VolumeSnapshotClass` used for creating volume snapshots or VM snapshots.
1. Similarly, Configure the **Backup Volume Snapshot Class Name**. This corresponds to the name of the `VolumeSnapshotClass` responsible for creating VM backups.
```

File: docs/advanced/csidriver.md [hunk 9]
```diff
@@ -0,0 +1,66 @@
...
+
+## Use the CSI driver
+
+After successfully configuring these settings, you can proceed to utilize the third-party StorageClass. You can apply the third-party StorageClass when creating an empty volume or adding a new block volume to a VM, enhancing your Harvester cluster's storage capabilities.
```
Comment (LucasSaintarbor) [inline docs/advanced/csidriver.md:64]:
```suggestion
After successfully configuring these settings, you can utilize the third-party StorageClass. You can apply the third-party StorageClass when creating an empty volume or adding a new block volume to a VM, enhancing your Harvester cluster's storage capabilities.
```

**Review by LucasSaintarbor — APPROVED** [review]
Small changes. LGTM 👍

File: docs/advanced/csidriver.md [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+sidebar_label: Third-Party CSI Driver
+title: "Third-Party CSI Driver"
+---
+
+<head>
+  <link rel="canonical" href="https://docs.harvesterhci.io/dev/advanced/csidriver"/>
+</head>
+
+
+Harvester now offers the capability to install a [Container Storage Interface (CSI)](https://kubernetes-csi.github.io/docs/introduction.html) in your Harvester cluster. This allows you to leverage external storage for the Virtual Machine's non-system data disk, allowing you to use different drivers tailored for specific needs, whether for performance optimization or seamless integration with your existing in-house storage solutions.
```
Comment (guangbochen) [inline docs/advanced/csidriver.md:12]:
```suggestion

_Available as of v1.2.0_

Harvester now offers the capability to install a [Container Storage Interface (CSI)](https://kubernetes-csi.github.io/docs/introduction.html) in your Harvester cluster. This allows you to leverage external storage for the Virtual Machine's non-system data disk, allowing you to use different drivers tailored for specific needs, whether for performance optimization or seamless integration with your existing in-house storage solutions.
```

File: docs/advanced/csidriver.md [hunk 11]
```diff
@@ -0,0 +1,63 @@
+---
+sidebar_position: 9
+sidebar_label: Third-Party CSI Driver
+title: "Third-Party CSI Driver"
```
Comment (guangbochen) [inline docs/advanced/csidriver.md:4]:
```suggestion
sidebar_label: Third-Party CSI Driver Support
title: "Third-Party CSI Driver Support"
```

File: docs/advanced/csidriver.md [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+
+Upgrading Harvester causes the changes to the OS in the `after-install-chroot` stage to be lost. You must also configure the `after-upgrade-chroot` to make your changes persistent across an upgrade. Refer to [Runtime persistent changes](https://rancher.github.io/elemental-toolkit/docs/customizing/runtime_persistent_changes/) before upgrading Harvester.
+
+:::
+
+## Install the CSI driver
+
+After installing the Harvester cluster is complete, refer to [How can I access the kubeconfig file of the Harvester cluster?](../faq.md#how-can-i-access-the-kubeconfig-file-of-the-harvester-cluster) to get the kubeconfig of the cluster.
+
+With the kubeconfig of the Harvester cluster, you can install the third-party CSI drivers into the cluster by following the installation instructions for each CSI driver. You must also refer to the CSI driver documentation to create the StorageClass and VolumeSnapshotClass in the Harvester cluster.
```
Comment (guangbochen) [inline docs/advanced/csidriver.md:47]:
```suggestion
With the kubeconfig of the Harvester cluster, you can install the third-party CSI drivers into the cluster by following the installation instructions for each CSI driver. You must also refer to the CSI driver documentation to create the `StorageClass` and `VolumeSnapshotClass` in the Harvester cluster.
```

File: docs/advanced/csidriver.md [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+1. Find and select **csi-driver-config**, and then select **⋮** > **Edit Setting** to access the configuration options.
+1. Set the **Provisioner** to the third-party CSI driver in the settings.
+1. Next, Configure the **Volume Snapshot Class Name**. This setting points to the name of the `VolumeSnapshotClass` used for creating volume snapshots or VM snapshots.
+1. Similarly, Configure the **Backup Volume Snapshot Class Name**. This corresponds to the name of the `VolumeSnapshotClass` responsible for creating VM backups.
+
+## Use the CSI driver
+
+After successfully configuring these settings, you can utilize the third-party StorageClass. You can apply the third-party StorageClass when creating an empty volume or adding a new block volume to a VM, enhancing your Harvester cluster's storage capabilities.
+
+With these configurations in place, your Harvester cluster is ready to make the most of the third-party storage integration.
```
Comment (guangbochen) [inline docs/advanced/csidriver.md:63]: Consider adding a new section of `## References` for KB articles.

File: docs/advanced/storageclass.md [hunk 14]
```diff
@@ -91,24 +91,6 @@ The `volumeBindingMode` field controls when volume binding and dynamic provision
 
 ![](/img/v1.2/storageclass/customize_tab_vol_binding_mode.png)
 
-## Multi Container Storage Interface support
```
Comment (guangbochen) [inline docs/advanced/storageclass.md:94]: prefer adding a note and referring to the new CSI doc page.

File: docs/advanced/csidriver.md (hunk 13 above)
Comment (guangbochen) [inline docs/advanced/csidriver.md:63]: maybe adding some screenshot if possible, thanks.

Comment (futuretea) [reply docs/advanced/csidriver.md]: Added

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**